### PR TITLE
feat: btc light client tests

### PR DESCRIPTION
**Pull request by vuvoth**
# Context 
-  `InsertHeader` logic test 
- `CleanUpfork` logic test

### CODE REVIEW

File: btc_light_client.go [hunk 1]
```diff
@@ -97,23 +100,23 @@ func (lc *BTCLightClient) forkOfBlockhash(bh chainhash.Hash) ([]*LightBlock, err
...
 }
 
-// TODO: We will do this in the next PR
+// We follow: 
 // - select the next finalize block base on 2 conditions:
 //   - this fork len greater than MaxForkAge
 //   - this fork is the most powerful fork
 //
-// - Remove all fork invalid
+// - Remove all forks invalid
```
Comment (robert-zaremba) [inline btc_light_client.go:108]:
```suggestion
// - Remove all invalid forks
```

File: btc_light_client.go [hunk 2]
```diff
@@ -192,7 +196,12 @@ func (b *BlockMedianTimeSource) Offset() time.Duration {
 
 func (lc *BTCLightClient) CheckHeader(parent wire.BlockHeader, header wire.BlockHeader) error {
 	noFlag := blockchain.BFNone
-	fork, _ := lc.forkOfBlockhash(parent.BlockHash())
+	
+	fork, err := lc.forkOfBlockhash(parent.BlockHash())
+
+	if err != nil {
```
Comment (robert-zaremba) [inline btc_light_client.go:202]:
```suggestion
	fork, err := lc.forkOfBlockhash(parent.BlockHash())
	if err != nil {
```

File: btc_light_client_store.go [hunk 3]
```diff
@@ -121,8 +120,15 @@ func (s *MemStore) SetIsNotHead(bh chainhash.Hash) {
 }
 
 // TODO: convert to iterator rather than returning a map
-func (s *MemStore) LatestBlockHashOfFork() map[chainhash.Hash]struct{} {
-	return s.latestBlockHashOfFork
+func (s *MemStore) LatestBlockHashOfFork() []chainhash.Hash{
+	hashes := []chainhash.Hash{}
+
```
Comment (robert-zaremba) [inline btc_light_client_store.go:125]:
```suggestion
```

File: btc_light_client_test.go [hunk 4]
```diff
@@ -1,9 +1,99 @@
...
+		btcHeader, _ := BlockHeaderFromHex(tc.header)
+		for id, str := range tc.headers {
+			h, _ := BlockHeaderFromHex(str)
+			decodedHeader[id] = h
+		}
+		lc := NewBTCLightClientWithData(&chaincfg.RegressionNetParams, decodedHeader, 0)
+
+		lcErr := lc.InsertHeader(btcHeader)
+
+		fmt.Println(lcErr)
```
Comment (robert-zaremba) [inline btc_light_client_test.go:53]:
```suggestion
		lc := NewBTCLightClientWithData(&chaincfg.RegressionNetParams, decodedHeader, 0)
		lcErr := lc.InsertHeader(btcHeader)
		fmt.Println(lcErr)
```

**Review by robert-zaremba — COMMENTED** [review]
few cosmetic suggestions

File: btc_light_client_store.go [hunk 5]
```diff
@@ -121,8 +120,14 @@ func (s *MemStore) SetIsNotHead(bh chainhash.Hash) {
...
-func (s *MemStore) LatestBlockHashOfFork() map[chainhash.Hash]struct{} {
-	return s.latestBlockHashOfFork
+func (s *MemStore) LatestBlockHashOfFork() []chainhash.Hash {
+	hashes := []chainhash.Hash{}
+	for h := range s.latestBlockHashOfFork {
+		hashes = append(hashes, h)
+	}
+
+	return hashes
+	// return s.latestBlockHashOfFork
```
Comment (sczembor) [inline btc_light_client_store.go:130]:
```suggestion
```

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**